From 202b1aabe95a1e8a826fd3f13545851bb8fea01c Mon Sep 17 00:00:00 2001
From: Sam Nelson <sam.nelson@ti.com>
Date: Mon, 3 Aug 2015 13:36:15 -0400
Subject: [PATCH 03/12] Documentation: ti,keystone-rproc: Add documentation for
 Keystone remote proc driver

- Documents device tree entries needed for the driver

Signed-off-by: Sam Nelson <sam.nelson@ti.com>
---
 .../bindings/remoteproc/linux,keystone-rproc.txt   | 75 ++++++++++++++++++++++
 1 file changed, 75 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/remoteproc/linux,keystone-rproc.txt

diff --git a/Documentation/devicetree/bindings/remoteproc/linux,keystone-rproc.txt b/Documentation/devicetree/bindings/remoteproc/linux,keystone-rproc.txt
new file mode 100644
index 0000000..62d3a75
--- /dev/null
+++ b/Documentation/devicetree/bindings/remoteproc/linux,keystone-rproc.txt
@@ -0,0 +1,75 @@
+Keystone Remoteproc driver
+
+This file provides information about, what the device tree entry for the
+remoteproc accessible remote processor should contain for keystone devices.
+
+Required properties:
+- compatible : 	"ti,keystone-rproc"
+- reg :			Register start address and the size that will be
+			used by the driver correspond to register names
+			under reg-names
+
+- reg-names = 		Should contain strings with the following names,
+			each representing a specific register.
+			"boot-address", "psc-mdstat", "psc-mdctl","ipcgr";
+
+- mem : 		Memory start address and size of the local memory
+			region of the remote processor used by the driver
+
+- interrupt-parent,
+  interrupts 	: 	Standard interrupt property for two interrupts
+			generated from the remote processor to the host
+			processor.
+			First entry:
+			Interrupt for vring  from remote processor
+			Second entry: (optional)
+			Interrupt for exception from remote processor
+
+- kick-gpio: 		Used to specify gpio device assocated with the
+			remote processor for the kick operation from Host
+			to the remote processor and should be defined as
+			per the bindings in
+			Documentation/devicetree/bindings/gpio/gpio.txt
+
+- clocks : 		desired clock domain property and should be defined
+			as per the bindings in
+			Documentation/devicetree/bindings/clock/keystone-gate.txt
+
+Optional properties:
+
+- memory-region:phandle to the reserved memory node to be associated
+			with the remoteproc device. The reserved memory node
+			should be a CMA memory node, and should be defined as
+			per the bindings in
+			Documentation/devicetree/bindings/reserved-memory/reserved-memory.txt
+
+Example:
+
+/* K2HK DSP node in SoC DTS file */
+soc {
+		dsp0: dsp0 {
+			compatible = "ti,keystone-rproc";
+			reg = <0x02620040 4
+			       0x0235083c 4
+			       0x02350a3c 4
+			       0x02620240 4>;
+			reg-names = "boot-address", "psc-mdstat", "psc-mdctl",
+				"ipcgr";
+			mem = <0x10e00000 0x00008000
+			       0x10f00000 0x00008000
+			       0x10800000 0x00100000>;
+			interrupt-parent = <&kirq0>;
+			interrupts = <8 0>;
+			kick-gpio = <&dspgpio0 27 0>;
+			clocks = <&clkgem0>;
+			status = "disabled";
+		};
+}
+
+
+&dsp0 {
+	status = "okay";
+	memory-region = <&dsp_common_cma_pool>;
+};
+
+
-- 
1.9.1

